File: 3354-make-array-elements-equal-to-zero/3354-make-array-elements-equal-to-zero.java
class Solution {
    public int countValidSelections(int[] nums) {
        int result = 0;
        int n = nums.length;
        int[] left = new int[n];
        int[] right = new int[n];

        for (int i = 1; i < n; ++i) {
            left[i] = left[i - 1] + nums[i - 1];
            right[n - i - 1] = right[n - i] + nums[n - i];
        }

        for (int i = 0; i < n; ++i) {
            if (nums[i] != 0) {
                continue;
            }
            if (left[i] == right[i]) {
                result += 2;
            }
            if (Math.abs(left[i] - right[i]) == 1) {
                result += 1;
            }
        }
        return result;
    }
}

File: 3354-make-array-elements-equal-to-zero/3354-make-array-elements-equal-to-zero.py
class Solution:
    def countValidSelections(self, nums: List[int]) -> int:
        n = len(nums)
        res = 0
        for i in range(n):
            if nums[i] != 0:
                continue
            
            for direction in [-1, 1]:  # -1 is left, 1 is right
                curr = i
                dir = direction
                tmp = nums[:]  # Make a copy so we don't mess up the original
                
                while 0 <= curr < n:
                    if tmp[curr] == 0:
                        curr += dir
                    elif tmp[curr] > 0:
                        tmp[curr] -= 1
                        dir *= -1      # reverse direction
                        curr += dir
                # At the end, check if all is zero
                if all(x == 0 for x in tmp):
                    res += 1
        return res